modules/dag-l0/src/main/scala/org/tessellation/dag/l0/Main.scala

```diff
@@ -24,18 +28,32 @@ import org.tessellation.schema.node.NodeState
 import org.tessellation.schema.semver.TessellationVersion
 import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshot}
 import org.tessellation.security.signature.Signed
+import org.tessellation.security.{Hasher, SecurityProvider}
 
 import com.monovore.decline.Opts
 import eu.timepit.refined.auto._
 
-object Main
+object Main extends DagL0Application {
+
+  override def mkGossipForkInfo[F[_]: Async: SecurityProvider: KryoSerializer: Hasher](
+    gossip: Gossip[F]
+  ): GossipForkInfo[F, GlobalSnapshotArtifact] =
+    GossipForkInfo.make[F, GlobalSnapshotArtifact](gossip)
+
+}
+
+abstract class DagL0Application
     extends TessellationIOApp[Run](
       name = "dag-l0",
       header = "Tessellation Node",
       version = TessellationVersion.unsafeFrom(BuildInfo.version),
       clusterId = ClusterId("6d7f1d6a-213a-4148-9d45-d7200f555ecf")
     ) {
 
+  def mkGossipForkInfo[F[_]: Async: SecurityProvider: KryoSerializer: Hasher](
```

Comment:
Are these context bounds necessary when there's no implementation?

Reply:
Yes. They are needed for the implementation and the signatures need to match for `override`s. Do you have something else in mind?

Reply:
I see. It does raise another question though:
Does it makes sense for that function to be typed in `F` when it's extending a class hard-coded to `IO` ?
Maybe the signature should be this:
```
def mkGossipForkInfo(
  gossip: Gossip[IO]
)(implicit SP: SecurityProvider[IO], K: KryoSerializer[IO], H: Hasher[IO])
 : GossipForkInfo[IO, GlobalSnapshotArtifact]
```

Reply:
Changed the signature to

```scala
def mkGossipForkInfo(
  gossip: Gossip[IO]
)(implicit SP: SecurityProvider[IO], K: KryoSerializer[IO], H: Hasher[IO])
 : GossipForkInfo[IO, GlobalSnapshotArtifact]
```